src/Token/Parser.php

```diff
@@ -105,7 +109,9 @@ private function parseClaims(string $data): array
                 continue;
             }
 
-            $claims[$claim] = $this->convertDate((string) $claims[$claim]);
+            $date = $claims[$claim];
+
+            $claims[$claim] = $this->convertDate(is_string($date) ? $date : json_encode($date, JSON_THROW_ON_ERROR));
```

Comment:
To make this explicit as well, the check is only needed for compatibility with previously issued tokens...